inst/include/hiv_demographic_projection.hpp

```diff
@@ -0,0 +1,168 @@
+  run_hiv_and_art_stratified_deaths_and_migration(time_step, pars, state_curr,
+                                                  state_next, working);
+}
+
+template <typename real_type>
+void run_hiv_ageing_and_mortality(int time_step,
+                                  const Parameters<real_type>& pars,
+                                  const State<real_type>& state_curr,
+                                  State<real_type>& state_next) {
+  // Non-hiv deaths
+  for (int g = 0; g < pars.num_genders; ++g) {
+    for (int a = 1; a < pars.age_groups_pop; ++a) {
+      state_next.hiv_natural_deaths(a, g) =
+          state_curr.hiv_population(a - 1, g) *
+          (1.0 - pars.survival(a, g, time_step));
+      state_next.hiv_population(a, g) = state_curr.hiv_population(a - 1, g);
+    }
+
+    // open age group
+    state_next.hiv_natural_deaths(pars.age_groups_pop - 1, g) +=
+        state_curr.hiv_population(pars.age_groups_pop - 1, g) *
+        (1.0 - pars.survival(pars.age_groups_pop, g, time_step));
+    state_next.hiv_population(pars.age_groups_pop - 1, g) +=
+        state_curr.hiv_population(pars.age_groups_pop - 1, g);
+  }
+}
+
+template <typename real_type>
+void run_hiv_and_art_stratified_aging(int time_step,
+                                      const Parameters<real_type>& pars,
+                                      const State<real_type>& state_curr,
+                                      State<real_type>& state_next) {
+  // TODO: better name for hiv_ag_prob - what is this? Prob of chance of someone
+  // getting HIV as a function of the size of the HIV population
+  Tensor2<real_type> hiv_ag_prob(pars.age_groups_hiv, pars.num_genders);
+  hiv_ag_prob.setZero();
+  // age coarse stratified HIV population
+  for (int g = 0; g < pars.num_genders; ++g) {
+    int a = pars.hiv_adult_first_age_group;
+    // Note: loop stops at age_groups_hiv-1; no one ages out of the open-ended
+    // age group
+    for (int ha = 0; ha < (pars.age_groups_hiv - 1); ++ha) {
+      for (int i = 0; i < pars.age_groups_hiv_span(ha); ++i) {
+        hiv_ag_prob(ha, g) += state_curr.hiv_population(a, g);
+        a++;
+      }
+      hiv_ag_prob(ha, g) =
+          (hiv_ag_prob(ha, g) > 0)
+              ? state_curr.hiv_population(a - 1, g) / hiv_ag_prob(ha, g)
+              : 0.0;
+    }
+
+    for (int ha = 1; ha < pars.age_groups_hiv; ++ha) {
+      for (int hm = 0; hm < pars.disease_stages; ++hm) {
+        state_next.hiv_strat_adult(hm, ha, g) =
+            (1.0 - hiv_ag_prob(ha, g)) *
+            state_curr.hiv_strat_adult(hm, ha, g);  // age-out
+        state_next.hiv_strat_adult(hm, ha, g) +=
+            hiv_ag_prob(ha - 1, g) *
+            state_curr.hiv_strat_adult(hm, ha - 1, g);  // age-in
+        if (time_step > pars.time_art_start)
+          for (int hu = 0; hu < pars.treatment_stages; hu++) {
+            state_next.art_strat_adult(hu, hm, ha, g) =
+                (1.0 - hiv_ag_prob(ha, g)) *
```

Comment:
These lines are not run atm. This is because in the example data (which has come from Jeff) `time_art_start` is 60 and max of `time_step` is 60 too so this is never run. I will add a test where I set `time_art_start` lower but just leaving a note here to ask Jeff about this when he is back